app/controllers/application_controller.rb

```diff
@@ -58,8 +57,10 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:pending_invite_email])
   end
 
-  def event_staff?(current_event)
-    current_user && current_event.teammates.where(user_id: current_user.id).any?
+  def event_staff?(event)
+    if event && current_user
+      event.teammates.where(user_id: current_user.id).any?
```

Comment:
Yeah, I stopped here but wanted to go further.  There's a lot of cruft conceptually around permissions and access logic.   The use of teammates vs reviewer_teammates vs reviewer_events is a great example.  event_policy.staff?, reviewer? and organizer?  are also on the chopping/refactoring block.n  I do want there to be a single source of this logic that the rest of the system can use.  

I saved it as a follow-up task until we've got critical functionality shipped.